### DISCUSSION TITLE
Reactive queries

### DISCUSSION THREAD

**Discussion by piovezanfernando** [ujjwalguptaofficial/JsStore#277] [General]
Hi, first thanks for the great component.
I have a question that I couldn't find anything about it, I use JSStore with Vue 2 my question is if there is a way to make the queries reactive to the point that when there is a new insertion in the table the data is updated on screen.
I have this small example in [CodeSandbox](https://codesandbox.io/s/try-relation-with-jsstore-forked-lfccc4?file=/src/App.vue) where a table is populated at creation and after I insert a new record through the button, but the data displayed on the screen is not updated.
I used another component that had an observer for this purpose, is the way to do the same for the JSStore? Or is there a simpler and faster way to solve this?

Thanks

Fernando Piovezan

**Comment by ujjwalguptaofficial** [comment]
The database are never reactive, but jsstore allows you flexibility to perform this. You can use jsstore middleware for this purpose - https://jsstore.net/tutorial/middleware/

so whenever insertion is successfull, you emit some events and listeners gets called causing reactivity.
